core/trino-spi/src/test/java/io/trino/spi/block/TestArrayBlockBuilder.java

```diff
@@ -62,12 +63,13 @@ public void testRetainedSizeInBytes()
         assertTrue(arrayBlockBuilder.getRetainedSizeInBytes() >= (expectedEntries * Long.BYTES + ClassLayout.parseClass(LongArrayBlockBuilder.class).instanceSize() + initialRetainedSize));
     }
 
-    @Test(expectedExceptions = IllegalStateException.class, expectedExceptionsMessageRegExp = "Expected current entry to be closed but was opened")
+    @Test
     public void testConcurrentWriting()
     {
         BlockBuilder blockBuilder = new ArrayBlockBuilder(BIGINT, null, EXPECTED_ENTRY_COUNT);
         BlockBuilder elementBlockWriter = blockBuilder.beginBlockEntry();
         elementBlockWriter.writeLong(45).closeEntry();
-        blockBuilder.appendStructure(new LongArrayBlockBuilder(null, 1).writeLong(123).closeEntry().build());
+        assertThatThrownBy(() -> blockBuilder.appendStructure(new LongArrayBlockBuilder(null, 1).writeLong(123).closeEntry().build()))
```

Comment:
If ` blockBuilder.appendStructure(new LongArrayBlockBuilder(null, 1)` does not throw, let's put it before `assertThatThrownBy`. Similarly, of `writeLong(123)` doesn't throw, let's keep it outside as well.

Reply:
.appendStructure is the one that throws.